### PR TITLE
fix(aws): Fix ram resources

### PR DESCRIPTION
**Pull request by bbernays**
fix(aws)!: Removed `aws_resource_share_associated_principals` and `aws_resource_share_associated_resources` relations of `aws_ram_resource_shares` and created a new top level table to hold this data `aws_resource_share_associations`.  To sync these resources, ensure to either use `tables: ["*"]` or `tables: ["aws_resource_share_associations"]`

fix(aws)!: Moved `aws_ram_resource_share_permissions` to be a relation of `aws_ram_resource_shares` instead of a top level table. To sync `aws_ram_resource_share_permissions` you'll need to ensure to sync `aws_ram_resource_shares` too by specifying either `tables: ["*"]` or `tables: ["aws_ram_resource_shares"]`

### CODE REVIEW

**Comment by erezrokah** [comment]
> BREAKING: RAM

Hi @bbernays, to show up in the changelog the format should be:
```
BREAKING CHANGE: Moved `aws_ram_resource_share_permissions` to be a relation of `aws_ram_resource_shares` instead of a top level table. To sync `aws_ram_resource_share_permissions` you'll need to ensure to sync `aws_ram_resource_shares` too by specifying either `tables: ["*"]` or `tables: ["aws_ram_resource_shares"]`
```
or something similar

**Comment by bbernays** [comment]
@erezrokah - Thank you I will be sure to change before merging... in the meanwhile I have added `no-automerge` tag

**Comment by erezrokah** [comment]
> @erezrokah - Thank you I will be sure to change before merging... in the meanwhile I have added `no-automerge` tag

Maybe also covert to draft until ready

File: plugins/source/aws/resources/services/ram/resource_share_associated_principals_fetch.go [hunk 1]
```diff
@@ -18,7 +18,6 @@ func fetchRamResourceShareAssociatedPrincipals(
 ) error {
 	input := &ram.GetResourceShareAssociationsInput{
 		AssociationType: types.ResourceShareAssociationTypePrincipal,
-		Principal:       parent.Item.(types.ResourceShare).ResourceShareArn,
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/ram/resource_share_associated_principals_fetch.go:21]: why do we skip this? in this case it's no longer a relation

File: plugins/source/aws/resources/services/ram/resource_share_associated_resources_fetch.go [hunk 2]
```diff
@@ -18,10 +18,9 @@ func fetchRamResourceShareAssociatedResources(
 ) error {
 	input := &ram.GetResourceShareAssociationsInput{
 		AssociationType: types.ResourceShareAssociationTypeResource,
-		ResourceArn:     parent.Item.(types.ResourceShare).ResourceShareArn,
-		MaxResults:      aws.Int32(500),
+		// ResourceArn:     parent.Item.(types.ResourceShare).ResourceShareArn,
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/ram/resource_share_associated_resources_fetch.go:21]: why do we skip this? in this case it's no longer a relation

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/aws/resources/services/ram/principals_fetch.go [hunk 3]
```diff
@@ -21,5 +23,19 @@ func fetchRamPrincipals(ctx context.Context, meta schema.ClientMeta, _ *schema.R
 		}
 		res <- response.Principals
 	}
+
+	input = &ram.ListPrincipalsInput{
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/ram/principals_fetch.go:27]: extract dup code?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/aws/resources/services/ram/resource_share_associated_principals_fetch.go (hunk 1 above)
Comment (bbernays) [reply plugins/source/aws/resources/services/ram/resource_share_associated_principals_fetch.go:21]: `ResourceShareArn` is not a a principal (accountID, orgid, user or role)....

File: plugins/source/aws/resources/services/ram/resource_share_associated_resources_fetch.go (hunk 2 above)
Comment (bbernays) [reply plugins/source/aws/resources/services/ram/resource_share_associated_resources_fetch.go]: It is now a top level table

File: plugins/source/aws/resources/services/ram/resource_share_associated_principals_fetch.go (hunk 1 above)
Comment (bbernays) [reply plugins/source/aws/resources/services/ram/resource_share_associated_principals_fetch.go:21]: It is now a top level table

**Review by candiduslynx — APPROVED** [review]

File: plugins/source/aws/resources/services/ram/resource_shares_mock_test.go [hunk 4]
```diff
@@ -20,10 +21,36 @@ func buildRamResourceSharesMock(t *testing.T, ctrl *gomock.Controller) client.Se
...
 
 	m.EXPECT().GetResourceShares(gomock.Any(), gomock.Any(), gomock.Any()).Return(
-		&ram.GetResourceSharesOutput{ResourceShares: []types.ResourceShare{object}}, nil)
+		&ram.GetResourceSharesOutput{ResourceShares: []types.ResourceShare{object}}, nil).MinTimes(1)
 
-	buildRamResourceShareAssociatedResourcesMock(t, m)
-	buildRamResourceShareAssociatedPrincipalsMock(t, m)
+	// buildRamResourceShareAssociatedResourcesMock(t, m)
+	// buildRamResourceShareAssociatedPrincipalsMock(t, m)
+
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ram/resource_shares_mock_test.go:28]:
```suggestion
```

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**